feature-libs/cart/base/components/clear-cart/clear-cart-dialog/clear-cart.service.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { TestBed } from '@angular/core/testing';
+import { ActiveCartFacade, OrderEntry } from '@spartacus/cart/base/root';
+import { Observable, of } from 'rxjs';
+import { ClearCartService } from './clear-cart.service';
+import { LaunchDialogService } from '@spartacus/storefront';
+import { GlobalMessageService } from '@spartacus/core';
+import createSpy = jasmine.createSpy;
+
+const mockCloseReason = 'Close Dialog';
+
+class MockGlobalMessageService implements Partial<GlobalMessageService> {
+  add = createSpy().and.stub();
+}
+
+class MockLaunchDialogService implements Partial<LaunchDialogService> {
+  closeDialog(_reason: string): void {}
+}
+
+class MockActiveCartFacade implements Partial<ActiveCartFacade> {
+  addEntry(_productCode: string, _quantity: number): void {}
+  getEntries(): Observable<OrderEntry[]> {
+    return of();
+  }
+  removeEntry(): void {}
+  isStable(): Observable<boolean> {
+    return of(true);
+  }
+}
+
+describe('ClearCartService', () => {
```

Comment:
Lets dicuss how to improve test coverage

Reply:
1 happy path per function with describe